### ISSUE TITLE
[Bug]: DTS passthrough issues on Android 12

### ISSUE THREAD

**Issue by branches78** [nova-video-player/aos-AVP#916]
### Problem description

TL;DR:

Some DTS streams are not passing through on Android 12, even when using the "force passthrough" option, they play correctly on Android 11, here's a comparison table:

![image](https://github.com/nova-video-player/aos-AVP/assets/87671962/87565297-b2ae-4c0e-9324-887055185c03)

### Your phone/tablet/androidTV model

Onn 2023 with Google TV

### Operating system version

Android 12

### Application version and app store

6.1.6-20230404.2327

### Additional system information

The device is connected directly to the AV receiver and the "Auto" option is selected in the device's settings while the option 2 is selected within the Nova app.

**Comment by branches78** [comment]
I tested the latest 6.2.18 pre-release version and the issue is still there.

**Comment by courville** [comment]
This is normal since if you look at the code difference nothing between 6.1.6 and 6.2.18 addresses the issue you are mentioning (and if it did I would have referred it on this issue).

Now can you please provide some details on the following questions:
- have you tested Android 11 and 12 on the same Onn device?
- can you detail the AV receiver you are using?

For clarification, "force passthrough" means that nova does not check HDMI capabilities and chooses to output stream even if the receiver does not support it which results in no sound or the correct sound if the receiver does not declare the support but supports it anyway (for licensing issues?). I am not saying it applies in your case since obviously your receiver is able to decode DTS and DTS HD.

**Comment by branches78** [comment]
Hello @courville and thank you for your reply.

> have you tested Android 11 and 12 on the same Onn device?

The Onn device comes with Android 12 out of the box, I tested the app on three other devices with Android 11:

- Homatics Box R 4K
- Homatics Box R 4K Plus
- Realme 4K Stick

On those 3 devices, all DTS streams passthrough correctly as stated in the above table.

> can you detail the AV receiver you are using?

Pioneer VSX-834, it supports everything, including Dolby Atmos and DTS:X.

It's important to clarify that all mentioned devices above only support DTS and DTS core (extracted from DTS-HD High Resolution Audio and DTS-HD Master Audio), it's just that it's not working on Android 12 for some reason (or at least on the Onn).

**Comment by courville** [comment]
Thanks for the clarifications.
What would be interesting, would be to understand if this is an Android 12 or a device specific issue.
For instance testing on a sabrina (Chromecast with Google TV) that is also amlogic based and running Android 12 would be a good reference point.
BTW, have you managed to get passthrough working on the Onn device using another video player (e.g. just player)?

@phhusson, would you be able to test on an Android 12 amlogic device on which you master the firmware?

**Comment by branches78** [comment]
> have you managed to get passthrough working on the Onn device using another video player

Yes, it works correctly with Emby, Jellyfin and GV Video Player (https://play.google.com/store/apps/details?id=com.gvuitech.videoplayer)

> @phhusson, would you be able to test on an Android 12 amlogic device on which you master the firmware?

If you want I can share the exact same test files I have so you can compare.

**Comment by branches78** [comment]
One of my devices (Homatics Box R 4K Plus) was recently updated to Android 12 beta (I'm a beta tester) and the same issue is also present, so this is a confirmation that the problem happens on Android 12 rather than on a single device (Onn 2023).

I hope you guys take a look at this issue.

**Comment by courville** [comment]
https://github.com/nova-video-player/aos-AVP/releases/tag/v6.2.25 available for testing

**Comment by branches78** [comment]
Hello, I just tested it and it finally works on my two Android 12 devices!.

Thank you so much for fixing this.
